src/backend/routers/scim.py

```diff
@@ -1,34 +1,72 @@
+import base64
 from typing import Optional
 
 from fastapi import APIRouter
+from fastapi.responses import JSONResponse
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.requests import Request
 
 import backend.crud.group as group_repo
 import backend.crud.user as user_repo
 from backend.config.routers import RouterName
-from backend.database_models import (
-    DBSessionDep,
-    User as DBUser,
-    Group as DBGroup,
-    UserGroup,
-)
+from backend.config.settings import Settings
+from backend.database_models import DBSessionDep
+from backend.database_models import Group as DBGroup
+from backend.database_models import User as DBUser
+from backend.database_models import UserGroup
 from backend.schemas.scim import (
-    ListUserResponse,
-    User,
-    Group,
-    CreateUser,
-    UpdateUser,
-    PatchUser,
-    PatchGroup,
     CreateGroup,
+    CreateUser,
+    Group,
     ListGroupResponse,
+    ListUserResponse,
+    PatchGroup,
+    PatchUser,
+    UpdateUser,
+    User,
 )
 from backend.services.logger.utils import get_logger
 
-router = APIRouter(prefix="/scim/v2")
+SCIM_PREFIX = "/scim/v2"
+scim_auth = Settings().auth.scim
+
+router = APIRouter(prefix=SCIM_PREFIX)
 router.name = RouterName.SCIM
 logger = get_logger()
 
 
+class SCIMMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        if not request.url.path.startswith(SCIM_PREFIX):
```

Comment:
is there a better way of handling the authentcation? sub apps? perhaps too complex